zero_bin/rpc/src/native/txn.rs

```diff
@@ -357,3 +398,228 @@ fn prestate_tracing_options(diff_mode: bool) -> GethDebugTracingOptions {
+                        .iter()
+                        .rev();
+                    let callee_raw = *evm_stack
+                        .nth(1)
+                        .expect("Stack must contain at least two values for a CALL instruction.");
+                    let lower_bytes = U160::from(callee_raw);
+                    Address::from(lower_bytes)
+                };
+
+                if precompiles().contains(&callee_address) {
+                    debug!("PRECOMPILE at address {} called.", &callee_address);
+                } else if callee_addr_to_code_hash.contains_key(&callee_address) {
+                    let code_hash = callee_addr_to_code_hash[&callee_address];
+                    call_stack.push((code_hash, next_ctx_available));
+                } else {
+                    // This case happens if calling an EOA. See also https://www.evm.codes/#00?fork=cancun
+                    debug!(
+                        "Callee address {} has no associated `code_hash`.",
+                        &callee_address
+                    );
+                }
+                next_ctx_available += 1;
+                prev_jump = None;
+            }
+            "JUMP" => {
+                let mut evm_stack = entry
+                    .stack
+                    .as_ref()
+                    .expect("No stack found in structLog.")
+                    .iter()
+                    .rev();
+                let jump_target = evm_stack
+                    .next()
+                    .expect("Stack must contain at least one value for a JUMP instruction.")
+                    .to::<u64>();
+                prev_jump = Some(jump_target);
+            }
+            "JUMPI" => {
+                let mut evm_stack = entry
+                    .stack
+                    .as_ref()
+                    .expect("No stack found in structLog.")
+                    .iter()
+                    .rev();
+                let jump_target = evm_stack
+                    .next()
+                    .expect("Stack must contain at least one value for a JUMP instruction.")
+                    .to::<u64>();
+                let jump_condition = evm_stack
+                    .next()
+                    .expect("Stack must contain at least two values for a JUMPI instruction.")
+                    .is_zero()
+                    .not();
+
+                prev_jump = if jump_condition {
+                    Some(jump_target)
+                } else {
+                    None
+                };
+            }
+            "JUMPDEST" => {
+                let (code_hash, ctx) = call_stack
+                    .last()
+                    .expect("Call stack was empty when a JUMPDEST was encountered.");
+                let jumped_here = if let Some(jmp_target) = prev_jump {
+                    debug_assert!(
+                        jmp_target == entry.pc,
+                        "The structlog seems to make improper JUMPs."
```

Comment:
I don't think this is something to be ever expected? did you even encounter it?

Reply:
I didn't.  Changed to an `ensure!` now.